src/metaxy/graph/status.py

```diff
@@ -0,0 +1,132 @@
+"""Feature metadata status inspection utilities.
+
+This module provides reusable SDK functions for inspecting feature metadata status,
+useful for both CLI commands and programmatic usage.
+"""
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, Any
+
+import narwhals as nw
+import polars as pl
+
+if TYPE_CHECKING:
+    from metaxy.metadata_store.base import MetadataStore
+    from metaxy.models.feature import BaseFeature
+
+
+def count_lazy_rows(lazy_frame: nw.LazyFrame[Any]) -> int:
+    """Return row count for a Narwhals LazyFrame.
+
+    Args:
+        lazy_frame: The LazyFrame to count rows from
+
+    Returns:
+        Number of rows in the LazyFrame
+    """
+    return lazy_frame.select(nw.len()).collect().to_polars()["len"].item()
+
+
+def preview_samples(
```

Comment:
Not sure if this helper is valuable, it just calls a few methods. The error handling it does is wrong IMO.